app/controllers/schools/participants_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class Schools::ParticipantsController < Schools::BaseController
+  skip_after_action :verify_authorized
+  skip_after_action :verify_policy_scoped
+
+  before_action :set_school_and_cohort
+  before_action :set_participant, except: :index
+
+  def index
+    @participants = User.order(:full_name).is_participant.in_school(@school.id)
+  end
+
+  def show
+    @mentor = @participant.early_career_teacher_profile&.mentor
+  end
+
+  def edit_details; end
+
+  def edit_mentor
+    @mentor_form = ParticipantForm.new(
+      mentor_id: @participant.early_career_teacher_profile.mentor&.id,
+    )
+
+    @mentors = @school.mentors.order(:full_name)
+  end
+
+  def update
+    @mentor_form = ParticipantForm.new(participant_form_params)
+
+    # mentor_id can be nil
+    if @mentor_form.mentor_id
+      mentor = @mentor_form.mentor_id != "later" && User.find(@mentor_form.mentor_id)
```

Comment:
Could we move it inside the form. Would be great to contain `"later"` trick within a single object.